pkg/mimirpb/prealloc_rw2_test.go

```diff
@@ -0,0 +1,463 @@
+// SPDX-License-Identifier: AGPL-3.0-only
```

Comment:
Look at the test `TestSplitWriteRequestByMaxMarshalSize_WriteRequestHasChanged` in this package. I think we want something similar for `FromWriteRequestToRW2Request()` too.